### ISSUE TITLE
dittoHeatmap scale colour display limits

### ISSUE THREAD

**Issue by gAleryani** [dtm2451/dittoSeq#90]
Hi, 

I first want to say what a great utility box! It really saved a lot of time making figures, thank you so much, ill be sure to acknowledge wherever I can :)

I wanted to enquire a bit more about the color gradient scale breaks for dittoheatmap. It would be nice if it had something similar to the display limit function of Seurat: "DoHeatmap(seurat_obj, disp.min = -2.5, disp.max = 2.5), to adjust for colour washing out for when you plot lowly divergent with highly divergent genes instead of having to go through pheatmap or complexheatmap function. Maybe that exists already and I missed it? My bad if thats the case!

**Comment by dtm2451** [comment]
Thanks!  I'm always happy to hear when people find dittoSeq helpful and I also certainly appreciate all citations/acknowledgements =).

You are correct that dittoHeatmap doesn't have a `disp.min`/`disp.max` equivalent currently.  Definitely an enhancement then, but it may not be too hard to make work... Probably not until at least next week, as I'm post-vacation and catching up on a lot of things at the moment, but I'll play around and see if I can implement something!

**Comment by gAleryani** [comment]
I hope you had a good break, I feel like scientists don't get enough of it.

Thank you for making things even easier for me in the future!

**Comment by chansigit** [comment]
o(30,15)
dittoHeatmap(ch.clean, genes=mbk_hm.genes, assay="RNA",slot="scale.data",
             heatmap.colors=colorRampPalette(c("blue", "white", "red"))(50),
             breaks=seq(-4, 4, length.out=50),
             annot.by = c("status", "anatomy", "mbk_clust"),
             order.by = "mbk_clust", cluster_rows=FALSE)

this is my solution

**Comment by dtm2451** [comment]
Took me far longer than I'd wanted to get back to you on this!

After some investigation, I've decided that I don't want to add these inputs to the function because doing so would require me to also add determination of the other side whenever users only give either `disp.min` or `disp.max`.  Seems simple, but that value calculation is dependent on scaling of the underlying data, and that's a step which dittoHeatmap actually leaves to the wrapped `pheatmap()` function.

Although I won't be adding it, I do want users to be able to achieve the desired effect.  As @chansigit notes, that can be achieved by giving the desired `disp.min`/`max` to a `breaks` input.  Just to expand the explanation a bit as this is going to serve as the "official" explanation of what to do...

## Here's how to do that:

Let's say this is the code that you used initially, and the plot output at the end:
```
dittoHeatmap(object, genes = my_genes, annot.by = "labels)
```
![Rplot01](https://user-images.githubusercontent.com/38870347/131926129-fec4b59c-4300-4c8d-87c1-da6616993085.png)

Now, add `breaks = seq(<desired-min>, <desired-max>, length.out = <length of colors>+1 )`, where:
- the default color palette has length of 50, so you'd put 51.
- the default color palette when `scaled.to.max = TRUE` has length of 25, so you would put 26.

So then for a desired `disp.min = -5` and `disp.max = 5`...
```
dittoHeatmap(object, genes = my_genes, annot.by = "labels,
    breaks = seq(-5, 5, length.out = 51)
    )
```
![Rplot02](https://user-images.githubusercontent.com/38870347/131926153-ea009f85-662a-45d5-ba56-6f3d6290b9ce.png)

**OR**

for desired a `disp.min = 0` and `disp.max = 0.65` with `scaled.to.max = TRUE`...
```
dittoHeatmap(object, genes = my_genes, annot.by = "labels,
    scaled.to.max = TRUE,
    breaks = seq(0, 0.65, length.out = 26)
    )
```
![Rplot03](https://user-images.githubusercontent.com/38870347/131926164-9dc5919d-e017-4021-9b09-634976debacc.png)

Also, MINOR NOTE that you may get a warning from `is.na()` being run on a vector with length > 1. This can safely be ignored, and will be fixed in an upcoming PR, #96.
